### ISSUE TITLE
Allow request with type overlay, to support event overlays 

### ISSUE THREAD

**Issue by POnakS** [DataDog/terraform-provider-datadog#1980]
### Terraform Version

1.5.0

### Affected Resource(s)

Please list the resources as a list, for example:

-   opc_instance
-   opc_storage_volume

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files

I would like to overlay events on my dashboard. When I configure overlay, terraform complains:
```
  widget {
    timeseries_definition {
      title = "My widget"
      request {
		/* request */
      }
      request { // new way which is not accepted by terraform provider
        display_type = "overlay"
        query {
          event_query {
            data_source = "events"
            indexes = ["*"]
            compute {
              aggregation = "count"
            }
            search {
              query = "My event"
            }
          }
        }
        event { // old way which does not work in datadog
          q = "MyEvent"
        }
      }
    }
```

(please note that I'm not trying to use both ways at same time)

### Expected Behavior

Overlay added to datadog:
![image](https://github.com/DataDog/terraform-provider-datadog/assets/90155580/231576d0-a8ec-4060-8ca1-3bf84ac3fc8a)

Event overlay generated, similar to this:
```
        {
            "type": "overlay",
            "response_format": "timeseries",
            "queries": [
                {
                    "data_source": "events",
                    "name": "query1",
                    "search": {
                        "query": "My event"
                    },
                    "indexes": [
                        "*"
                    ],
                    "compute": {
                        "aggregation": "count"
                    },
                    "group_by": []
                }
            ],
            "formulas": [
                {
                    "formula": "query1"
                }
            ]
        }
```

### Actual Behavior

If I try to use `event {` block, terraform json will be not valid and won't display overlay. I need to go to the UI, and paste again my query to make it work (so datadog UI provides correct event overlay).

If I try to use new overlay, which would produce correct json, it complains that:
```
│ Error: Invalid enum value
│
│   with datadog_dashboard.my-dashboard,
│   on datadog_module.tf line 293, in resource "datadog_dashboard" "my-dashboard":
│  293:       request {
│
│ `invalid value 'overlay' for WidgetDisplayType: valid values are [area bars line]`
╵
```
### Steps to Reproduce

1. Define dd dashboard with timeseries_definition.
2. Add events to dashboard
3. `terraform apply`
4. Verify if events are visible in dashboard

### References

Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:

-   GH-1234

**Comment by nkzou** [comment]
I'll forward your feature request to support the new overlay format in the dashboard resource. In the meantime, can you confirm that the `event {}` block is under the `timeseries_definition` block, and not the `requests` block like in your example?

**Comment by POnakS** [comment]
Yes, that's right, this is just mistake in the example. However this event block does not generate valid json after apply.

**Comment by nkzou** [comment]
Could you provide an example of the terraform definition and the resulting invalid json?

**Comment by POnakS** [comment]
I mean json is valid as per json spec, but datadog won't display events based on this json.


Terraform code:
```
  widget {
    timeseries_definition {
      title = "My widget"
      request {
		/* request */
      }
      event { // old way which does not work in datadog
          q = "MyEvent"
      }
    }
  }
```

This is json which is visible in UI after apply:
```
{
    "viz": "timeseries",
    "requests": [
        {
               // query drawing lines here
        },
        {
            "type": "overlay",
            "response_format": "timeseries",
            "queries": [
                {
                    "data_source": "events",
                    "name": "query1",
                    "indexes": [
                        "*"
                    ],
                    "compute": {
                        "aggregation": "count"
                    },
                    "group_by": [],
                    "search": {
                        "query": "MyEvent"
                    }
                }
            ],
            "formulas": [
                {
                    "formula": "query1"
                }
            ],
            "events": [
                {
                    "q": "MyEvent",
                    "tags_execution": "and"
                }
            ],
        }
```

And this is result in UI
![image](https://github.com/DataDog/terraform-provider-datadog/assets/90155580/f8d5795c-05e1-4c92-891c-044aecbeb186)
When you edit this by hand (even to the same value) and save, it will change to following json:
```
{
    "viz": "timeseries",
    "requests": [
        {
               // query drawing lines here
        },
        {
            "type": "overlay",
            "response_format": "timeseries",
            "queries": [
                {
                    "data_source": "events",
                    "name": "query1",
                    "indexes": [
                        "*"
                    ],
                    "compute": {
                        "aggregation": "count"
                    },
                    "group_by": [],
                    "search": {
                        "query": "MyEvent"
                    }
                }
            ],
            "formulas": [
                {
                    "formula": "query1"
                }
            ]
        }
     ],
     // remaining widget definition
}
```

So basically this is old syntax which is updated by UI on save, but not updated by API and therefore terraform module does not work.

**Comment by nkzou** [comment]
Hi again, unfortunately I haven't been able to reproduce this issue - using the old format on our testing environment displays events as expected. Could you open a support ticket at https://help.datadoghq.com/? We believe this is a client-side rendering issue. You can mention the DataViz team to get things routed faster - the support engineer will be able to see exactly what's going wrong in your account.

**Comment by POnakS** [comment]
Thanks!

**Comment by tmonck** [comment]
https://github.com/DataDog/datadog-api-client-go/issues/2088 I was having a similar issue not sure it was 100% the same issue but I opened up an issue on the sdk as that's where the problem lives I believe. https://github.com/DataDog/datadog-api-client-go/issues/2088
